pipx/main.py

```diff
@@ -6,20 +6,27 @@
 import argparse
 import functools
 import logging
+from pkg_resources import parse_version
 import shlex
 import sys
 import textwrap
 import urllib.parse
 from typing import Dict, List
 
 import argcomplete  # type: ignore
-
-from . import commands, constants
 from .colors import bold, green
+from . import commands
+from . import constants
 from .util import PipxError, mkdir
 from .Venv import VenvContainer
 
+
 __version__ = "0.14.0.0"
+_, __version_info__, subver, *_ = parse_version(__version__)._key  # type: ignore
+if isinstance(subver, tuple):
+    while len(__version_info__) < 4:
+        __version_info__ += (0,)
+    __version_info__ += subver
```

Comment:
would you mind adding a unit test for this?

Reply:
Sure. Are we set on only allowing a version with 1 to 4 components, plus an optional a/b/c component?

Reply:
Yeah it will always be something like 0.1.2.3.4, 0.1.2.3.4b0, 0.1.2.3.4rc0, etc.